api/v1alpha1/progressivesync_types.go

```diff
@@ -57,6 +57,17 @@ type Stage struct {
 	// Targets is the targets to sync in the stage
 	//+kubebuilder:validation:Optional
 	Targets Targets `json:"targets,omitempty"`
+
+	// SyncOptions is the options for sync operation executed for the stage
+	//+kubebuilder:validation:Optional
+	SyncOptions SyncOptions `json:"syncOptions,omitempty"`
```

Comment:
Do we want to keep this at the stage level? I'm asking because it feels a little heavy, because normally one should prune or not prune.

Reply:
Was thinking that it might bring some more flexibility like you might want to enable pruning only in staging stages but not in production for some reason... I am completely fine to move this to `type ProgressiveSyncSpec struct` 
I don't have strong opinion on it, WDYT?

Reply:
I would go for a more generic option to start lean, and then we can expand it if we find a valid use case.

Reply:
@maruina hey, sorry for delay, I've just moved syncOpts to the PS level so it will be applied for all stages.